src/coinjoin/coinjoin.cpp

```diff
@@ -120,8 +120,8 @@ bool CCoinJoinBroadcastTx::CheckSignature(const CBLSPublicKey& blsPubKey) const
 bool CCoinJoinBroadcastTx::IsExpired(const CBlockIndex* pindex, const llmq::CChainLocksHandler& clhandler) const
 {
     // expire confirmed DSTXes after ~1h since confirmation or chainlocked confirmation
-    if (nConfirmedHeight == -1 || pindex->nHeight < nConfirmedHeight) return false; // not mined yet
-    if (pindex->nHeight - nConfirmedHeight > 24) return true; // mined more than an hour ago
+    if (!nConfirmedHeight.has_value() || pindex->nHeight < *nConfirmedHeight) return false; // not mined yet
+    if (pindex->nHeight - *nConfirmedHeight > 24) return true; // mined more than an hour ago
```

Comment:
Speaking of magic numbers ... 24 seems pretty magical :magic_wand: 
> if (pindex->nHeight - *nConfirmedHeight > 24)...

Reply:
24 is old code, not new, should not be a blocker for this pull request IMHO

Reply:
> 24 is old code, not new, should not be a blocker for this pull request IMHO

Yeah, I was mostly joking around about that one. Sorry to be confusing. :slightly_smiling_face: